plugins/processors/gpuattributes/internal/awsneuron_metric_modifier.go

```diff
@@ -0,0 +1,328 @@
+		m := slice.At(i)
+
+		dps := m.Sum().DataPoints()
+		for j := 0; j < dps.Len(); j++ {
+			attributes := dps.At(j).Attributes()
+			attributes.RemoveIf(func(label string, value pcommon.Value) bool {
+				_, exists := MetricAttributesToKeep[label]
+				if !exists {
+					return true
+				}
+				return false
+			})
+		}
+	}
+}
+
+// This method prefixes NeuronCore and NeuronDevice values with `core` and `device` respectively
+// to make the attribute values more verbose
+func prefixCoreAndDeviceLabels(originalMetric pmetric.Metric) {
+	dps := originalMetric.Sum().DataPoints()
+	for i := 0; i < dps.Len(); i++ {
+		dp := dps.At(i)
+		for attributeKey, attributeValuePrefix := range attributeValuePrefixingMap {
+			if value, exists := dp.Attributes().Get(attributeKey); exists {
+				dp.Attributes().PutStr(attributeKey, attributeValuePrefix+value.AsString())
+			}
+		}
+	}
+}
+
+// This method performs selective duplication of a metric based on the types for which duplication needs to be performed.
+// A metric is duplicated for pod and container only if pod correlation has been done successfully
+func (md *AwsNeuronMetricModifier) duplicateMetrics(metricsSlice pmetric.MetricSlice, originalMetricName string, originalMetricDatapoints pmetric.NumberDataPointSlice) pmetric.MetricSlice {
+	newMetricsSlice := pmetric.NewMetricSlice()
+	metricModifications := metricModificationsMap[originalMetricName]
+
+	// check if pod correlation has been performed, if not then don't emit metric for container and pod
+	duplicateForNodeOnly := false
+	podName, exists := originalMetricDatapoints.At(0).Attributes().Get(PodName)
+	if !exists || len(podName.Str()) == 0 {
+		duplicateForNodeOnly = true
+	}
+
+	for i := 0; i < metricsSlice.Len(); i++ {
+		metric := metricsSlice.At(i)
+		if duplicateForNodeOnly {
+			duplicateMetricForType(metric, containerinsightscommon.TypeNode, originalMetricName).CopyTo(newMetricsSlice.AppendEmpty())
+		} else {
+			for _, prefix := range metricModifications.DuplicationTypes {
+				duplicateMetricForType(metric, prefix, originalMetricName).CopyTo(newMetricsSlice.AppendEmpty())
+			}
+		}
+	}
+
+	return newMetricsSlice
+}
+
+// This method creates new metrics by prefixing the metric name with each k8 concepts (pod, node and container).
+// It also adds logTypes to all the metric datapoint attributes.
+func duplicateMetricForType(metric pmetric.Metric, duplicateType string, originalMetricName string) *pmetric.Metric {
+	metricCopy := pmetric.NewMetric()
+	metric.CopyTo(metricCopy)
+	metricCopy.SetName(strings.ToLower(duplicateType) + "_" + metricCopy.Name())
+
+	datapoints := metricCopy.Sum().DataPoints()
+	for i := 0; i < datapoints.Len(); i++ {
+		datapoints.At(i).Attributes().PutStr(containerinsightscommon.MetricType, duplicateType+logTypeSuffix+metricModificationsMap[originalMetricName].LogTypeSuffix)
+	}
+
+	return &metricCopy
+}
+
+func getMetricWithMetadata(metric pmetric.Metric, name string, unit string) pmetric.Metric {
```

Comment:
you are setting metadata not getting them

Reply:
updating the name to setMetricMetadata